examples/miscellaneous/plot_outlier_detection_bench.py

```diff
@@ -0,0 +1,193 @@
+"""
+==========================================
+Evaluation of outlier detection estimators
+==========================================
+
+This example benchmarks outlier detection algorithms, Local Outlier
+Factor (LOF) and Isolation Forest (IForest), using ROC curves on
+classical anomaly detection datasets. The algorithm performance
+is assessed in an outlier detection context:
+
+1. The algorithms are trained on the whole dataset which is assumed to
+contain outliers.
+
+2. The ROC curve is computed on the same dataset using the knowledge
+of the labels.
+"""
+
+# Author: Pharuj Rajborirug <pharuj.ra@kmitl.ac.th>
+# License: BSD 3 clause
+
+print(__doc__)
+
+# %%
+# Define a data preprocessing function
+# ----------------------------------
+#
+# The example uses real-world datasets available in `scikit-learn.datasets`
+# and the sample size of some datasets to speed up computation.
+# After the data preprocessing, the datasets' targets will have two classes,
+# 0 representing inliers and 1 representing outliers. The preprocess_dataset
+# function returns data and target.
+
+import numpy as np
+from sklearn.datasets import fetch_kddcup99, fetch_covtype, fetch_openml
+from sklearn.preprocessing import LabelBinarizer
+import pandas as pd
+
+rng = np.random.RandomState(42)
+
+
+def preprocess_dataset(dataset_name="http"):
+
+    # loading and vectorization
+    print(f"Loading {dataset_name} data")
+    if dataset_name in ["http", "smtp", "SA", "SF"]:
+        dataset = fetch_kddcup99(subset=dataset_name, percent10=True, random_state=rng)
+        X = dataset.data
+        y = dataset.target
+    if dataset_name == "SF":
+        idx = rng.choice(X.shape[0], int(X.shape[0] * 0.1), replace=False)
+        X = X[idx]  # reduce the sample size
+        y = y[idx]
+        lb = LabelBinarizer()
+        x1 = lb.fit_transform(X[:, 1].astype(str))
+        X = np.c_[X[:, :1], x1, X[:, 2:]]
+        y = (y != b"normal.").astype(int)
+    if dataset_name == "SA":
+        idx = rng.choice(X.shape[0], int(X.shape[0] * 0.1), replace=False)
+        X = X[idx]  # reduce the sample size
+        y = y[idx]
+        lb = LabelBinarizer()
+        x1 = lb.fit_transform(X[:, 1].astype(str))
+        x2 = lb.fit_transform(X[:, 2].astype(str))
+        x3 = lb.fit_transform(X[:, 3].astype(str))
+        X = np.column_stack((X[:, :1], x1, x2, x3, X[:, 4:]))
+        y = (y != b"normal.").astype(int)
+    if dataset_name == "http" or dataset_name == "smtp":
+        y = (y != b"normal.").astype(int)
```

Comment:
Maybe you can factorize some lines of code by inserting these if statements inside the more general 
`if dataset_name in ["http", "smtp", "SA", "SF"]:`
statement, i.e similarly to what you do for the individual cases in your larger  
`if dataset_name in ["glass", "wdbc", "cardiotocography"]:`
statement.

Reply:
@ArturoAmorQ  Could you check? I think it should be good now